File: gaia-sdk-java/gaia-sdk-graphql/src/main/kotlin/gaia/sdk/request/type/Behaviour.kt
package gaia.sdk.request.type

import gaia.sdk.client.Type
import gaia.sdk.request.intf.*
import gaia.sdk.client.Input
import gaia.sdk.Uuid
import gaia.sdk.ISO8601
import gaia.sdk.Struct
import gaia.sdk.request.input.*
import gaia.sdk.request.enumeration.*

/**
 * this type represents the behaviour information
 */
class Behaviour: Type() {

    /**
     * The behaviour id
     */
    fun identityId() { 
        add {"identityId" } 
    }

    /**
     * The behaviour reference id
     */
    fun reference() { 
        add {"reference" } 
    }

    /**
     * The name of the behaviour
     */
    fun qualifier() { 
        add {"qualifier" } 
    }

    /**
     * Detailed description about the behaviour
     */
    fun appendent() { 
        add {"appendent" } 
    }

    /**
     * The list of labels of the behaviour
     */
    fun labellist() { 
        add {"labellist" } 
    }

    /**
     * The behaviour xml
     */
    fun behaviour() { 
        add {"behaviour" } 
    }
}

File: gaia-sdk-python/gaia_sdk/graphql/response/type/Behaviour.py


from dataclasses import dataclass
from typing import List
Uuid = str
String = str
ISO8601 = str
Struct = dict
Float = float
from gaia_sdk.graphql.request.enumeration.RuntimeState import RuntimeState
from gaia_sdk.graphql.request.enumeration.SkillState import SkillState

@dataclass
class Behaviour:
    """
    this type represents the behaviour information
    """
    dictionary: dict
    """
    The behaviour id
    """
    @property
    def identity_id(self) -> Uuid:
        return Uuid(self.dictionary.get("identityId"))
    """
    The behaviour reference id
    """
    @property
    def reference(self) -> Uuid:
        return Uuid(self.dictionary.get("reference"))
    """
    The name of the behaviour
    """
    @property
    def qualifier(self) -> String:
        return String(self.dictionary.get("qualifier"))
    """
    Detailed description about the behaviour
    """
    @property
    def appendent(self) -> String:
        return String(self.dictionary.get("appendent"))
    """
    The list of labels of the behaviour
    """
    @property
    def label_list(self) -> List[String]:
        return list(map(lambda x: String(x), self.dictionary.get("labelList")))
    """
    The behaviour xml
    """
    @property
    def behaviour(self) -> String:
        return String(self.dictionary.get("behaviour"))
